13-TeV-examples/uproot_python/ttbar_analysis.ipynb

```diff
@@ -0,0 +1,1236 @@
+        "        \"mtop2\": ak.to_list(data[data_type]['mtop_1']),\n",
+        "        \"mtop3\": ak.to_list(data[data_type]['mtop_2'])\n",
+        "    })\n",
+        "\n",
+        "    main_analysis_df = pd.concat([temp_df, main_analysis_df],ignore_index=True)\n",
+        "\n",
+        "    # Process data for \"MC\" sample\n",
+        "    data_type = \"MC\"\n",
+        "    temp_df = pd.DataFrame({\n",
+        "        \"met_et\": ak.to_list(data[data_type]['Met_et']),\n",
+        "        \"Nbjet\": ak.to_list(data[data_type]['b_jets']),\n",
+        "        \"Njet\": ak.to_list(data[data_type]['q_jets']),\n",
+        "        \"pt\": ak.to_list(data[data_type]['pt_jets']),\n",
+        "        \"mW\": ak.to_list(data[data_type]['W_mass']),\n",
+        "        \"Weight\": ak.to_list(data[data_type]['total_weight']),\n",
+        "        \"mtop\": ak.to_list(data[data_type]['mtop']),\n",
+        "        \"mtop2\": ak.to_list(data[data_type]['mtop_1']),\n",
+        "        \"mtop3\": ak.to_list(data[data_type]['mtop_2'])\n",
+        "    })\n",
+        "\n",
+        "    mc_data_df = pd.concat([temp_df, mc_data_df],ignore_index=True)\n",
+        "\n",
+        "    # Process data for \"Single top\" sample\n",
+        "    data_type = \"Single top\"\n",
+        "    temp_df = pd.DataFrame({\n",
+        "        \"met_et\": ak.to_list(data[data_type]['Met_et']),\n",
+        "        \"Nbjet\": ak.to_list(data[data_type]['b_jets']),\n",
+        "        \"Njet\": ak.to_list(data[data_type]['q_jets']),\n",
+        "        \"pt\": ak.to_list(data[data_type]['pt_jets']),\n",
+        "        \"mW\": ak.to_list(data[data_type]['W_mass']),\n",
+        "        \"Weight\": ak.to_list(data[data_type]['total_weight']),\n",
+        "        \"mtop\": ak.to_list(data[data_type]['mtop']),\n",
+        "        \"mtop2\": ak.to_list(data[data_type]['mtop_1']),\n",
+        "        \"mtop3\": ak.to_list(data[data_type]['mtop_2'])\n",
+        "    })\n",
+        "\n",
+        "    single_top_data_df = pd.concat([temp_df, single_top_data_df],ignore_index=True)\n",
+        "\n",
+        "    # Process data for \"Diboson\" sample\n",
+        "    data_type = \"Diboson\"\n",
+        "    temp_df = pd.DataFrame({\n",
+        "        \"met_et\": ak.to_list(data[data_type]['Met_et']),\n",
+        "        \"Nbjet\": ak.to_list(data[data_type]['b_jets']),\n",
+        "        \"Njet\": ak.to_list(data[data_type]['q_jets']),\n",
+        "        \"pt\": ak.to_list(data[data_type]['pt_jets']),\n",
+        "        \"mW\": ak.to_list(data[data_type]['W_mass']),\n",
+        "        \"Weight\": ak.to_list(data[data_type]['total_weight']),\n",
+        "        \"mtop\": ak.to_list(data[data_type]['mtop']),\n",
+        "        \"mtop2\": ak.to_list(data[data_type]['mtop_1']),\n",
+        "        \"mtop3\": ak.to_list(data[data_type]['mtop_2'])\n",
+        "    })\n",
+        "\n",
```

Comment:
Again, I'm not against this strategy, but couldn't we setup a simple `for` loop that just encloses these steps into one?